core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -76,6 +74,7 @@ class SnappySession(@transient private val sc: SparkContext,
 
   self =>
 
+  SnappyContext.initConf = sc.conf
```

Comment:
Is this required after the initialization in global block? Probably the issue is that clear is not being done.